include/usb/usb_device.h

```diff
@@ -143,6 +143,20 @@ struct usb_ep_cfg_data {
 	uint8_t ep_addr;
 };
 
+/**
+ * @brief USB Interface container
+ *
+ * This structure represents interface related data
+ * requred for USB core stack part to correctly address
+ * the requests. Right now interface container provides
+ * only one alternate setting, this may change in the future.
+ */
+struct usb_if_container {
+	const struct usb_if_descriptor *const iface;
```

Comment:
This well be reworked in subsequent patches as discussed, right? (just to clarify)

Reply:
Yes. I don't want to make everything in one PR, as it will grow significantly. This is jus a base, 'usb_if_container' will be reworked anyway. I treat 'topic-usb' branch as wip/devel branch.

Reply:
Yes. I don't want to make everything in one PR, as it will grow significantly. This is jus a base, 'usb_if_container' will be reworked anyway. I treat 'topic-usb' branch as wip/devel branch.